src/ExecBackends/ExecSpace.hpp

```diff
@@ -400,6 +404,33 @@ class ExecSpace
     return TransferImpl<MEMBACKEND, EXECPOLICIES, MEMBACKEND, EXECPOLICIES, T, I>::do_it(p_dest, *this, p_src, *this, n);
   }
 
+  /**
+   * Copy `n` elements of the array `p_src` to the `p_dest` array.
+   * 
+   * @pre `p_src` and `p_dest` should be allocated so that they can hold at least 
+   * `n` elements.
+   * @pre `p_dest` should be managed by the memory backend of `this`.
+   * @pre `p_src` should be managed by the memory backend of `ms`.
+   */
+  template<class MEMSRC, class EXEPOLSRC, typename T, typename I>
+  inline bool copy(T* p_dest, const T* p_src, const I& n, const ExecSpace<MEMSRC,EXEPOLSRC>& ms) const
```

Comment:
I think there is some confusion here. None of the `copy` should be `const`. Original methods should work.